### PR TITLE
[CPU] Avoid using cache for constant inplace or multi-child edges

### PR DESCRIPTION
**Pull request by EgorDuplensky**
To avoid concurrent allocation while resolving inplace conflicts
-------
### Tickets:
 - 79095

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@maxnick Please review the PR

File: src/plugins/intel_cpu/src/edge.cpp [hunk 1]
```diff
@@ -292,7 +292,10 @@ void MKLDNNEdge::externalAllocate(MKLDNNWeightsSharing::Ptr weightsCache) {
     if (status != Status::NeedAllocation)
         return;
 
-    if (weightsCache) {
+    bool isTheOnlyChildEdge = getParent()->getChildEdges().size() == 1;
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/edge.cpp:295]: Actually we need to check that there are peer edges on the same port. If the port of the edge is different the memory will not be shared.

Comment (EgorDuplensky) [reply src/plugins/intel_cpu/src/edge.cpp]: changed to
``` c++
getParent()->getChildEdgesAtPort(getInputNum()).size() == 1;
```

**Review by maxnick — APPROVED** [review]

**Comment by EgorDuplensky** [comment]
Change is being benchmarked at the moment

File: src/plugins/intel_cpu/src/edge.cpp [hunk 2]
```diff
@@ -288,12 +288,44 @@ std::string MKLDNNEdge::name() const {
...
 
+
+
 void MKLDNNEdge::externalAllocate(MKLDNNWeightsSharing::Ptr weightsCache) {
+    auto isInPlace = [](const MKLDNNNodePtr& node, int port) -> bool {
+        if (node->inplace == MKLDNNNode::InPlaceType::InPlace) {
+            return true;
+        } else if (node->inplace == MKLDNNNode::InPlaceType::NoInPlace) {
+            return false;
+        }
```
[the remark this answers is not in the document]
Comment (EgorDuplensky) [reply src/plugins/intel_cpu/src/edge.cpp]: Discussed, updated according to the discussion.
Extra checks have been removed

**Comment by EgorDuplensky** [comment]
Release branch PR:
- #10795

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**